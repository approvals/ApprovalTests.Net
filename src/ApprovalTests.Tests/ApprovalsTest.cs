using System.Collections.Generic;
using System.IO;
using ApprovalUtilities.Utilities;
using NUnit.Framework;

namespace ApprovalTests.Tests
{
    [TestFixture]
    public class ApprovalsTest
    {
        private static readonly string[] text = {"abc", "123", "!@#"};

        [Test]
        public void Text()
        {
            Approvals.Verify("should be approved");
        }

        [Test]
        public void EnumerableWithLabel()
        {
            Approvals.VerifyAll(text, "collection");
        }

        [Test]
        public void TestExistingFile()
        {
            var path = PathUtilities.GetDirectoryForCaller();
            var copy = path + "copyOfa.txt";
            File.Copy(path + "a.txt", copy, true);
            Approvals.VerifyFile(copy);
        }

        [Test]
        public void TestBytes()
        {
            var path = PathUtilities.GetDirectoryForCaller();
            Approvals.VerifyBinaryFile(File.ReadAllBytes(path + "a.png"), "png");
        }


        [Test]
        public void EnumerableWithLabelAndFormatter()
        {
            Approvals.VerifyAll(text, "collection", t => "" + t.Length);
        }

        [Test]
        public void EnumerableWithHeaderAndFormatter()
        {
            var word = "Llewellyn";
            Approvals.VerifyAll(word, word.ToCharArray(), c => c + " => " + (int) c);
        }

        [Test]
        public void VerifyAllWithNull()
        {
            var words = new[] {"abc", null, "123", "!@#"};
            Approvals.VerifyAll(words, "");
        }

        [Test]
        public void VerifyAllNull()
        {
            string[] words = null;
            Approvals.VerifyAll(words, "words");
        }

        [Test]
        public void VerifyNullDictionary()
        {
            Dictionary<string, int> words = null;
            Approvals.VerifyAll(words);
        }

        [Test]
        public void DictionarySimple()
        {
            Approvals.VerifyAll(FireFlyMap());
        }

        [Test]
        public void Dictionary()
        {
            Approvals.VerifyAll("Firefly", FireFlyMap());
        }

        [Test]
        public void DictionaryCustom()
        {
            Approvals.VerifyAll("Firefly", FireFlyMap(), (k, v) => $"\"{k}\" => {v}");
        }

        [Test]
        public void DictionaryCustomNoHeader()
        {
            Approvals.VerifyAll(FireFlyMap(), (k, v) => $"\"{k}\" => {v}");
        }

        private static Dictionary<string, string> FireFlyMap()
        {
            return new Dictionary<string, string>
            {
                {"Caption", "Mal"},
                {"2nd In Command", "Zoey"},
                {"Pilot", "Wash"},
                {"Companion", "Inara"},
                {"Muscle", "Jayne"},
                {"Mechanic", "Kaylee"},
                {"Doctor", "Simon"},
                {"Pastor", "Book"},
                {"Stowaway", "River"}
            };
        }

        [Test]
        public void EnumerableWithFormatter()
        {
            Approvals.VerifyAll(text, t => "" + t.Length);
        }

        [Test]
        public void JsonText()
        {
            var json = @"{""GivenNames"":""John"",""FamilyName"":""Smith"",""Spouse"":""Jill"",""Address"":{""Street"":""1 Puddle Lane"",""Suburb"":null,""Country"":""USA""},""Children"":[""Sam"",""Mary""],""Title"":""Mr""}";
            Approvals.VerifyJson(json);
        }
    }
}