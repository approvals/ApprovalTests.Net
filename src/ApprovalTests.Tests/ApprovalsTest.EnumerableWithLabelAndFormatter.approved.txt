collection[0] = 3
collection[1] = 3
collection[2] = 3
